src/backend/src/lib.rs

```diff
@@ -154,36 +155,63 @@ pub struct Config {
     pub ecdsa_key_name: String,
     // A list of allowed callers to restrict access to endpoints that do not particularly check or use the caller()
     pub allowed_callers: Vec<Principal>,
+    pub supported_credentials: Option<Vec<SupportedCredential>>,
+    /// Root of trust for checking canister signatures.
+    pub ic_root_key_raw: Option<Vec<u8>>,
+}
+
+fn update_config(arg: InitArg) {
+    mutate_state(|state| {
+        let ic_root_key_raw = match arg
+            .ic_root_key_der
+            .map(|der_key| extract_raw_root_pk_from_der(&der_key))
+        {
+            Some(Ok(root_key)) => Some(root_key),
```

Comment:
There is no enforcement in place to ensure that no root_key args is passed when the canister is deployed on mainnet, is that safe enough?

Reply:
I believe so, but I wonder whether we should ask for a security review after this PR?

Reply:
Maybe. Generally speaking we have been keeping them in the loop of Oisy each time we have build a new feature.

Reply:
I'll do that then!

Reply:
I'll also point to this PR as well.